tests/deepsparse/v2/unit/test_text_generation.py

```diff
@@ -0,0 +1,387 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+import numpy
+from transformers import AutoTokenizer
+
+import pytest
+from deepsparse.transformers.helpers import get_deployment_path
+from deepsparse.transformers.pipelines.text_generation import TextGenerationInput
+from deepsparse.transformers.utils import DecoderKVCache
+from deepsparse.transformers.utils.helpers import initialize_kv_cache_state
+from deepsparse.v2 import InferenceState, PipelineState
+from deepsparse.v2.text_generation import (
```

Comment:
One general comment: while those tests do test (duh!), I think they could do a better job at their second (arguably, equally important) task, which is being expressive and guiding fellow engineers (most likely not familiar with the module) through the logic of how components ought to behave. I recommend pretending that you are new to TextGeneration-thing and adding comments that could facilitate the understanding of what's going on in those tests.

Reply:
Yah I can add docstrings/explanations for the tests

Reply:
done